app.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Created by dusting on 10/26/15.
+ */
+var getFile = require('./getFile');
+
+var investigation = getFile.processArgs(process);
+
+var graphDict = {};
+var graphDictArray = [];
+var touchedDict = {seen: [], starts:[], ends:[]};
+var pathDict = {paths:[]};
+
+witnessAccounts(investigation, findPaths);
+
+
+//This section maps nodes and edges to an object
+function witnessAccounts(investigation, callback) {
+    for (var people = 0; people < investigation.length; people++) {
+        mapNodes(investigation[people]);
+    }
+    callback();
+}
+
+function mapNodes(events){
+
+    for (var i = 0; i<events.length; i++){
+
+        var node = events[i];
+        var edge = events[i+1];
+        if (graphDict.hasOwnProperty(node)){
+            addEdges(node, edge, i, events.length);
+        }else{
+            addNode(node);
+            addEdges(node, edge, i, events.length);
+        }
+    }
+}
+
+function addNode(v){
+    graphDict[v] = []; //Initialize a vertex with empty edges
+    graphDictArray.push(v);
+}
+
+function addEdges(node, edge, count, uBound){
+    if (edge != undefined && count <= uBound){
+        graphDict[node].push(edge);
+    }
+}
+
+
+//This section follows the edges from the root and creates the path arrays
+function findPaths() {
```

Comment:
You are going to need to keep track of valid starts and valid ends. You may have witness accounts where the start is include in another account, but is not the start for that account. That means that the vertice is not a start. You have to track this for ends as well.

Once you know which paths you need to start from (find paths from) and which nodes you need to end (find paths to). It's trivial to look up algorithm for path finding on wikipedia.

What you want is the longest path bet ween each start and each end.